test_scripts/MobileProjection/Defects/common.lua

```diff
@@ -0,0 +1,191 @@
+  :Do(function(_, data)
+      m.hmi.getConnection():SendResponse(data.id, data.method, "SUCCESS", { })
+    end)
+end
+
+function m.stopStreaming(pAppId)
+  if m.streamingStatus[pAppId] == true then
+    m.mobile.getSession(pAppId):StopStreaming(m.streamFiles[pAppId])
+    m.log("App " .. pAppId .. " stops streaming")
+    m.streamingStatus[pAppId] = false
+  else
+    utils.cprint(33, "Streaming is unable to stop since it's not started")
+  end
+end
+
+function m.registerApp(pAppId)
+  if not pAppId then pAppId = 1 end
+  local session = m.mobile.createSession(pAppId)
+  session:StartService(7)
+  :Do(function()
+      local corId = session:SendRPC("RegisterAppInterface", m.app.getParams(pAppId))
+      m.hmi.getConnection():ExpectNotification("BasicCommunication.OnAppRegistered")
+      :Do(function(_, d)
+          m.app.setHMIId(d.params.application.appID, pAppId)
+        end)
+      session:ExpectResponse(corId, { success = true, resultCode = "SUCCESS" })
+      function session:ExpectEndService(pServiceId)
+        local event = m.run.createEvent()
+        event.matches = function(_, data)
+          return data.frameType == constants.FRAME_TYPE.CONTROL_FRAME and
+            data.serviceType == pServiceId and
+            data.sessionId == self.sessionId and
+            data.frameInfo == constants.FRAME_INFO.END_SERVICE
+        end
+        return session:ExpectEvent(event, "End Service Event")
+      end
+      function session:SendEndServiceAck(pServiceId)
+        self:Send({
+          frameType = constants.FRAME_TYPE.CONTROL_FRAME,
+          serviceType = pServiceId,
+          frameInfo = constants.FRAME_INFO.END_SERVICE_ACK
+        })
+      end
+    end)
+  return session:ExpectNotification("OnHMIStatus", { hmiLevel = "NONE" })
+end
+
+function m.checkTimeout(pBaseParam, pVerifiedParam, pExpTS, pTollerance)
+  if not pTollerance then pTollerance = m.tollerance end
+  if m.ts[pBaseParam] then
+    if m.ts[pVerifiedParam] == nil then
+      m.run.fail(pVerifiedParam .. " is not received")
+    else
+      local delay = math.abs(m.ts[pVerifiedParam] - m.ts[pBaseParam])
+      if delay > pExpTS + pTollerance then
+        m.run.fail("Delay between `" .. pBaseParam .. "` and `" .. pVerifiedParam .. "` is too high, expected ~ "
+          .. pExpTS .. "ms(+" .. pTollerance .. "), actual " .. delay .. "ms")
+      end
+      if delay < pExpTS - pTollerance then
+        m.run.fail("Delay between `" .. pBaseParam .. "` and `" .. pVerifiedParam .. "` is too low, expected ~ "
+          .. pExpTS .. "ms(-" .. pTollerance .. "), actual " .. delay .. "ms")
+      end
+    end
+  end
+end
+
+function m.checkSequence(pConType, pExpSeq)
+  local function isArraysEqual(pTbl1, pTbl2)
+    if #pTbl1 ~= #pTbl2 then return false end
+    for k in pairs(pTbl1) do
+      if pTbl1[k] ~= pTbl2[k] then return false end
+    end
+    return true
+  end
+  local seq = m.seq[pConType]
+  if not isArraysEqual(pExpSeq, seq) then
```

Comment:
I propose to use the existed function `isTableEqual` from utils instead of the` isArraysEqual`.

Reply:
@GetmanetsIrina Function `isTableEqual` does not check the order of keys in table, hence it is not help us to verify sequences.
E.g. in case of t1 = { "a", b" } and t2 = { "b", a" } it returns `true`